File: common/child_pipe_test_slave.py
import parent_pipe


class TestServer(parent_pipe.StdioServer):
    """
    These are used in test_child_pipe.js
    """
    def rpc_test1(self, v):
        return v+1

    def rpc_test2(self, a, b, c):
        assert a == 'abc'
        assert b == 'def'
        assert c['ghi'] == 'jkl'
        return [a, b, c], 'foo'

    def rpc_testerr(self):
        raise ValueError('testerr always raises this error')

if __name__ == '__main__':
    ts = TestServer()
    ts.run()

File: common/child_pipe_test_slave.js
'use strict';
const _ = require('lodash');
const async = require('async');
const assert = require('assert');
const parent_pipe = require('./parent_pipe');
const logio = require('../common/logio');

function main() {
  let pp = new parent_pipe.ParentJsonPipe({}, {
    rpc_test1: (v, cb) => {
      cb(null, v+1);
    },
    rpc_test2: (a, b, c, cb) => {
      assert.equal(a, 'abc');
      assert.equal(b, 'def');
      assert.equal(c.ghi, 'jkl');
      cb(null, [[a, b, c], 'foo']);
    },
    rpc_testerr: (cb) => {
      cb(new Error('testerr always raises this error'));
    },
  });
}

main();
